server.js

```diff
@@ -617,31 +617,49 @@ app.post('/submitanswer', function(req, res) {
             return res.status(400).send('Could not find the question');
         }
 
-        logger.log(common.formatString('User {0} attempted to answer question {1} with "{2}"', [userId, questionId, answer]));
+        // check if question is locked for the student
+        /*if(req.session.user.type !== common.userTypes.ADMIN && questions.isUserLocked(userId, question)){
+            return res.status(423).send('Sorry question is Locked, please try again later');
+        }*/
+        questions.isUserLocked(userId, question, function(isLocked, waitTime){
+            console.log(waitTime);
+            if (isLocked){
+                return res.status(423).send(waitTime);
+            }
 
-        var value = questions.verifyAnswer(question, answer);
-        var points = question.points;
-        var text = value ? 'correct' : 'incorrect';
-        var status = value ? 200 : 500;
-        var response = {text: text, points: points};
+            logger.log(common.formatString('User {0} attempted to answer question {1} with "{2}"', [userId, questionId, answer]));
 
-        if (req.session.user.type === common.userTypes.ADMIN) {
-            return res.status(status).send(response);
-        }
+            var value = questions.verifyAnswer(question, answer);
+            var points = question.points;
+            var text = value ? 'correct' : 'incorrect';
```

Comment:
Doesn't look right that we are checking if `value` is defined twice in a row, this should be combined like
``` JavaScript
if (value) {
    //Do stuff
}
```